### DISCUSSION TITLE
Storybook 8 Release Plan

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25276] [Announcement]
Hello Storybook Community,

Storybook 8 is underway! This release contains fewer breaking changes than the previous major releases, and we are prioritizing a smooth migration experience from 7.x to 8.0. A work-in-progress [migration guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-7x-to-800) is available to assist you in preparing for the changes.

We will be tracking work items in this [burndown board](https://github.com/orgs/storybookjs/projects/19/views/7). Below is the timeline we are working towards. We will update this if anything were to change.  In the meantime, you can try out the latest Storybook 8 prerelease:

```
npx storybook@next upgrade --prerelease
```

### ✅ Alpha 

**Shipped on: November 28, 2023**

This is the initial preview of Storybook 8. During this phase, expect frequent prereleases with new features and breaking changes. 

### ✅ Beta 

**Shipped on: Feburary 2, 2024**

Once we are feature complete, Storybook 8 Beta will be released. This release incorporates all breaking changes and new features in 8.0; the documentation and migration guide will also be complete by then. The focus during the beta phase is on refinements and bug squashing. We encourage users to start migrating on their own projects at this phase; any feedback and bug reports will be very, very helpful!

### ✅ Release Candidate

**Shipped on: February 29, 2024**

At the Release Candidate (RC) stage, Storybook 8 should be stable enough for users to migrate to in production. We will coordinate a round of public QA with the Storybook community, addressing any final bugs as we collectively test the upgrade on our projects.

### ✅ Stable

**Shipped on: March 11, 2024**

Finally, we will be ready to ship Storybook 8 stable release!

**Comment by ghost** [comment]
Yeah!

**Comment by oenu** [comment]
Is the RC still on track?

**Reply by vanessayuenn** [reply]
Updated the target ship dates for upcoming RC and Stable!

**Comment by davidglezz** [comment]
There is a typo, it should be 2024.
https://storybook.js.org/releases/8.0

![image](https://github.com/storybookjs/storybook/assets/194784/6fb04d17-6bb1-43c6-9fd7-e22e8daebbab)

**Reply by davidglezz** [reply]
Fixed

**Comment by oberruti** [comment]
`RC: Target ship date: February 26, 2024` Is this finally happening today?

**Reply by oenu** [reply]
Looks like it was bumped to the 29th

**Reply by vanessayuenn** [reply]
@oberruti @oenu indeed we were at the nail-biting phase of a release. But RC has just been shipped today! 🚀

**Comment by levrik** [comment]
I'm starting a new project today. Would you recommend going with Storybook 8 Beta directly or to better stay away until it's released. Given that it's currently less than 2 weeks away, I'm unsure but I'm also seeing in this thread that Beta and RC both have been moved several times in the past so I'm unsure what to pick.

**Reply by vanessayuenn** [reply]
RC has just been shipped today! We are quite confident in its stability so I would recommend starting your new project with 8.0-RC. This way you don't have to deal with migration soon down the line.

**Reply by levrik** [reply]
Thanks for your answer @vanessayuenn. Bootstrapping with RC now.

**Comment by hatschusnies** [comment]
It seems like the toolbar is still using icons from the icons component. How to integrate @storybook/icons to the toolbar?

**Comment by willwill96** [comment]
I tried upgrading our app to the 8 rc, but there are honestly a ton of deprecations for this release. Will have to postpone trying to upgrade to another time

**Reply by vanessayuenn** [reply]
can you share which deprecations were dealbreakers for you to upgrade?

**Reply by willwill96** [reply]
Actually a lot of the things I initially thought were daunting were covered by the migration guide 😄 

The only major issue remaining is some flakiness in our storybook composition setup.

With the setup I'm testing, storybook 8 is the main storybook, then there are 2 storybook 7 and 2 storybook 6 storybooks that are listed under composition. (Not sure whether versions are relevant but figured I would mention them)

When storybook loads, it seems like there's about a 50/50 chance whether the composed storybooks load or not. When they don't load, a loading animation is stuck like below:

![screenshot_2024-03-02_at_4 35 47___pm](https://github.com/storybookjs/storybook/assets/10997562/2409e9d4-6fcd-4747-9389-ee2e41398243)


From what I can tell, nothing relevant is output to the browser console. 

I'll try to put together a repro in the meantime

**Reply by willwill96** [reply]
I put together a reproduction of the error here:
https://github.com/willwill96/Storybook8CompositionRepro

If you pull down the repo, all you need to do is build and run the docker image and then navigate to http://localhost:3003/main-storybook and you should see that it always fails to load at least one of the compositional storybooks

Edit: created a github issue for the error here: ;https://github.com/storybookjs/storybook/issues/26288
